commec/regulation/containers.py

```diff
@@ -0,0 +1,383 @@
+    Returns None if it cannot be determined.
+    """
+    if isinstance(identifier, int):
+        return AccessionFormat.TAXID
+
+    patterns = {
+        AccessionFormat.UNIPROT: re.compile(
+            r"^(?:[OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z][0-9](?:[A-Z][A-Z0-9]{2}[0-9]){1,2})$"
+        ),
+        AccessionFormat.TAXID: re.compile(
+            r"^[0-9]+$"
+        ),
+        AccessionFormat.GENBANK: re.compile(
+            r"^(?:[A-Z][0-9]{5}|[A-Z]{2}[0-9]{6,8}|[A-Z]{3}[0-9]{5,7})(?:\.\d+)?$"
+        ),
+    }
+
+    for fmt, pattern in patterns.items():
+        if pattern.fullmatch(identifier):
+            return fmt
+
+    return None
+
+@dataclass(frozen=True)
+class Accession:
+    """
+    A unified hashable identifier for 
+    taxonomy IDs, GenBank, or UniProt accessions.
+
+    Used for hashed index accession for rapid annotations retrieval
+    from a pandas DataFrame.
+    """
+    code: str
+    type: AccessionFormat   # one of: "taxid", "genbank", "uniprot"
+
+    def __init__(self, accession: Optional[str]=None,
+                       taxid: Optional[int]=None,
+                       uniprot: Optional[str]=None,
+                       genbank: Optional[str]=None,
+                       row : pd.Series = None):
+        # Best guess if unknown input.
+        if accession:
+            acc_fmt = derive_accession_type(accession)
+            if acc_fmt:
+                object.__setattr__(self, "code", accession)
+                object.__setattr__(self, "type", acc_fmt)
+                return
+
+        # ensure only one is set
+        values = [(taxid, AccessionFormat.TAXID),
+                  (uniprot, AccessionFormat.UNIPROT),
+                  (genbank, AccessionFormat.GENBANK)]
+        non_empty = [(v, t) for v, t in values if not pd.isna(v)]
+        if len(non_empty) == 0:
+            error_id = row.to_string() if not row is None else "[Unknown row info]"
+            logger.debug("Accession [\n%s\n] must have at least one non-empty"
+                         " value for tax_id, genbank, or uniprot.\n"
+                         "It has been assigned to TaxID 0.", error_id)
+            non_empty = [("0", AccessionFormat.TAXID)]
+
+        object.__setattr__(self, "code", str(non_empty[0][0]))
+        object.__setattr__(self, "type", non_empty[0][1])
+
+    def get(self) -> tuple[str, str]:
+        """Return (code, type)."""
+        return self.code, self.type
+
+    def __str__(self):
+        return f"{self.code}"
+
+    def __repr__(self):
+        return f"{self.type}:{self.code}"
+
+@dataclass
+class Region:
+    """
+    Name and Acronym for regional information as either 
+    a country, or wider a organisation:
+    e.g. European Union, EU
+    e.g. New Zealand, NZ
+    """
+    name : str = ""
+    acronym : str = ""
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        return self.name + " ["+self.acronym+"]"
+
+class ListMode(StrEnum):
```

Comment:
```suggestion
class ControlMode(StrEnum):
    """
    Describes whether to treat Taxonomy IDs on a control list as regulated.
    Typically depends on the regional context.
    
    * REGULATED - All taxids from this list are regulated.
    * CONSENSUS - Taxids are regulated if they appear in more than 1 of these lists.
    * IGNORE - Ignore this list entirely.
    """
    REGULATED = "Regulated"
    CONSENSUS = "Used to determine default regulation"
    IGNORE = "Ignored"
```

Agree that we do want to support an "Ignore" option for lists from outside a region, e.g. could have a `--region US` vs. `--region US --only` CLI options